### PR TITLE
Fix DQN target update frequency

### LINKED ISSUE #322: Target network isn't updated to the correct frequency when `target_network_frequency % train_frequency != 0`
Because this `if` statement

https://github.com/vwxyzjn/cleanrl/blob/c37a3ec4ef8d33ab7c8a69d4d2714e3817739365/cleanrl/dqn.py#L205

is inside this one

https://github.com/vwxyzjn/cleanrl/blob/c37a3ec4ef8d33ab7c8a69d4d2714e3817739365/cleanrl/dqn.py#L185

Consequently, the target network is updated when `global_step % train_frequency == 0 and global_step % target_network_frequency == 0`.

For example, when you run

```shell
python cleanrl/dqn.py --target-network-frequency 501
```

The target network is updated every 5010 timesteps, not every 501 timesteps.

### PR DESCRIPTION
**Pull request by qgallouedec**
## Description

Closes #322 

## Types of changes

- [x] Bug fix
- [ ] New feature
- [ ] New algorithm
- [ ] Documentation

## Checklist:

- [x] I've read the [CONTRIBUTION](https://github.com/vwxyzjn/cleanrl/blob/master/CONTRIBUTING.md) guide (**required**).
- [x] I have ensured `pre-commit run --all-files` passes (**required**).
- [ ] I have updated the documentation and previewed the changes via `mkdocs serve`.
- [ ] I have updated the tests accordingly (if applicable).

If you are adding new algorithm variants or your change could result in performance difference, you may need to (re-)run tracked experiments. See https://github.com/vwxyzjn/cleanrl/pull/137 as an example PR. 
- [ ] I have contacted [vwxyzjn](https://github.com/vwxyzjn) to obtain access to the [openrlbenchmark W&B team](https://wandb.ai/openrlbenchmark) (**required**).
- [ ] I have tracked applicable experiments in [openrlbenchmark/cleanrl](https://wandb.ai/openrlbenchmark/cleanrl) with `--capture-video` flag toggled on (**required**).
- [ ] I have added additional documentation and previewed the changes via `mkdocs serve`.
    - [ ] I have explained note-worthy implementation details.
    - [ ] I have explained the logged metrics.
    - [ ] I have added links to the original paper and related papers (if applicable).
    - [ ] I have added links to the PR related to the algorithm variant.
    - [ ] I have created a table comparing my results against those from reputable sources (i.e., the original paper or other reference implementation).
    - [ ] I have added the learning curves (in PNG format).
    - [ ] I have added links to the tracked experiments.
    - [ ] I have updated the overview sections at the [docs](https://docs.cleanrl.dev/rl-algorithms/overview/) and the [repo](https://github.com/vwxyzjn/cleanrl#overview)
- [ ] I have updated the tests accordingly (if applicable).

### CODE REVIEW

File: cleanrl/dqn.py [hunk 1]
```diff
@@ -201,9 +201,9 @@ def linear_schedule(start_e: float, end_e: float, duration: int, t: int):
             loss.backward()
             optimizer.step()
 
-            # update the target network
-            if global_step % args.target_network_frequency == 0:
-                target_network.load_state_dict(q_network.state_dict())
+        # update the target network
+        if global_step % args.target_network_frequency == 0:
+            target_network.load_state_dict(q_network.state_dict())
```
Comment (vwxyzjn) [inline cleanrl/dqn.py:206]: Maybe change it to 

```
        # update the target network
        if global_step > args.learning_starts and global_step % args.target_network_frequency == 0:
            target_network.load_state_dict(q_network.state_dict())
```

**Review by vwxyzjn — CHANGES_REQUESTED** [review]
@qgallouedec thanks so much for the fix!

File: cleanrl/dqn.py (hunk 1 above)
Comment (qgallouedec) [reply cleanrl/dqn.py:206]: What about something like this instead:

```python
if global_step > learning_starts :
    if global_step % train_frequency == 0:
        ...

    if global_step % target_network_frequency == 0:
        ...
```

I find it more readable.

Comment (vwxyzjn) [reply cleanrl/dqn.py:206]: That would work great.

**Review by vwxyzjn — APPROVED** [review]
LGTM. Given that this change does not impact the performance of our algorithm variants using their default parameters, we do not need to re-run the benchmark for these variants.

**Pull request merged**